pkg/sysregistriesv2/shortnames.go

```diff
@@ -0,0 +1,261 @@
+// corresponding field is left empty.
+const defaultShortNameMode = types.ShortNameModePermissive
+
+// userShortNamesFile is the user-specific registries.conf file to store aliases.
+var userShortNamesFile = filepath.FromSlash(".config/containers/short-name-aliases.conf")
+
+// shortNameAliasConf is a subset of the `V2RegistriesConf` format.  It's used in the
+// software-maintained `userShortNamesFile`.
+type shortNameAliasConf struct {
+	// A map for aliasing short names to their fully-qualified image
+	// reference counter parts.
+	Aliases map[string]string `toml:"aliases"`
+
+	namedAliases map[string]reference.Named
+}
+
+// ResolveShortNameAlias performs an alias resolution of the specified name.
+// The user-specific short-name-aliases.conf has precedence over aliases in the
+// assembled registries.conf.  It returns whether the provided name could be
+// resolved to an alias, the possibly resolved alias, and an error.
+//
+// Note that the specified name is taken as it without normalizing it.
+func ResolveShortNameAlias(ctx *types.SystemContext, name string) (bool, reference.Named, error) {
+	if err := validateShortName(name); err != nil {
+		return false, nil, err
+	}
+	confPath, lock, err := shortNameAliasConfPathAndLock(ctx)
+	if err != nil {
+		return false, nil, err
+	}
+
+	// Acquire the lock as a reader to allow for multiple routines in the
+	// same process space to read simultaneously.
+	lock.RLock()
+	defer lock.Unlock()
+
+	aliasConf, err := loadShortNameAliasConf(confPath)
+	if err != nil {
+		return false, nil, err
+	}
+
+	// Note: short-name-alias.conf has precedence.
+	value, resolved := aliasConf.namedAliases[name]
+	if resolved {
+		return resolved, value, nil
+	}
+
+	config, err := getConfig(ctx)
+	if err != nil {
+		return false, nil, err
+	}
+	value, resolved = config.namedAliases[name]
+	return resolved, value, nil
+}
+
+// editShortNameAlias loads the aliases.conf file and changes it. If value is
+// set, it adds the name-value pair as a new alias. Otherwise, it will remove
+// name from the config.
+func editShortNameAlias(ctx *types.SystemContext, name string, value *string) error {
+	if err := validateShortName(name); err != nil {
+		return err
+	}
+	if value != nil {
+		if _, err := parseAlias(*value); err != nil {
+			return err
+		}
+	}
+	confPath, lock, err := shortNameAliasConfPathAndLock(ctx)
+	if err != nil {
+		return err
+	}
+
+	// Acquire the lock as a writer to prevent data corruption.
+	lock.Lock()
+	defer lock.Unlock()
+
+	// Load the short-name-alias.conf, add the specified name-value pair,
+	// and write it back to the file.
+	conf, err := loadShortNameAliasConf(confPath)
+	if err != nil {
+		return err
+	}
+
+	if value != nil {
+		conf.Aliases[name] = *value
+	} else {
+		// If the name does not exist, throw an error.
+		if _, exists := conf.Aliases[name]; !exists {
+			return errors.Errorf("short-name alias %q not found in %q: please check registries.conf files", name, confPath)
```

Comment:
Should this be even more helpful, listing the currently configured alias?

Reply:
The full UI should probably differentiate between “no alias defined at all” and “alias defined, but not in the editable file”.

Reply:
I defer to future improvements for this one.